W-16300511 Adding more information on types in json moduleloader and gaps with f… (#390)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

.Output:
----
"ADDRESS"
----

=== Built-in JSON Schema Types

DataWeave supports the following JSON schema types transformed into DataWeave types:

[cols="2,2",options="header"]
|===
| JSON Schema Type | DataWeave Type
| `number`| `Number`
| `integer`| `Number`
| `string`| `String`
| `boolean`| `Boolean`
| `array`| `Array`
| `enums`| `Union`
| `null`| `Null`
| `object`| `Object`
|===
 
=== Limitations

> * Formats on JSON schema string types aren't supported and are transformed to `String` without any validation or restriction.

Reviewer: ```suggestion
* DataWeave doesn't support formats on JSON schema string types and transforms these types to `String` without any validation or restriction.
```

> * Restrictions on number types such as maximums or minimums aren't supported and are be transformed to `Number` without any validation or restriction.

Reviewer: ```suggestion
* DataWeave doesn't support restrictions on number types such as maximums or minimums and transforms these types to `Number` without any validation or restriction.
```

> * Patterns aren't supported.

Reviewer: ```suggestion
* DataWeave doesn't support patterns.
```

=== Object types

> Non-required properties in JSON Schema object types are marked as optional properties in the DataWeave Object type. 

Reviewer: ```suggestion
DataWeave marks non-required properties in JSON Schema object types as optional properties in the DataWeave Object type. 
```
